monitor/service.go

```diff
@@ -110,16 +123,71 @@ func (m *Monitor) Open() error {
 }
 
 // Close closes the monitor system.
-func (m *Monitor) Close() {
+func (m *Monitor) Close() error {
+	if !m.open() {
+		m.Logger.Println("Monitor is already closed.")
+		return nil
+	}
+
 	m.Logger.Println("shutting down monitor system")
+	m.mu.Lock()
 	close(m.done)
+	m.mu.Unlock()
 
 	m.wg.Wait()
+
+	m.mu.Lock()
 	m.done = nil
+	m.mu.Unlock()
+
 	m.DeregisterDiagnosticsClient("build")
 	m.DeregisterDiagnosticsClient("runtime")
 	m.DeregisterDiagnosticsClient("network")
 	m.DeregisterDiagnosticsClient("system")
+	return nil
+}
+
+// SetGlobalTag can be used to set tags that will appear on all points
+// written by the Monitor.
+func (m *Monitor) SetGlobalTag(key string, value interface{}) {
+	m.mu.Lock()
+	m.globalTags[key] = fmt.Sprintf("%v", value)
+	m.mu.Unlock()
+}
+
+// RemoteWriterConfig represents the configuration of a remote writer
+type RemoteWriterConfig struct {
+	RemoteAddr string
+	NodeID     string
+	Username   string
+	Password   string
+	ClusterID  uint64
+}
+
+// SetRemoteWriter can be used via and RPC call to set a remote location
+// for writing monitoring information.
+func (m *Monitor) SetRemoteWriter(c RemoteWriterConfig) error {
+	// Ignore the monitor's config settings.
+	m.mu.Lock()
+	m.storeEnabled = true
+	m.storeInterval = DefaultStoreInterval
+	m.storeDatabase = DefaultStoreDatabase
+	m.mu.Unlock()
```

Comment:
Because we decided that the  influxdb.monitor config should be ignored when influx is running in the environments that use this new mechanism.